lib/chef-dk/policyfile/chef_server_lock_fetcher.rb

```diff
@@ -0,0 +1,107 @@
+require "chef-dk/policyfile_lock"
+require "chef-dk/exceptions"
+
+module ChefDK
+  module Policyfile
+    class ChefServerLockFetcher
+
+      attr_accessor :name
+      attr_accessor :source_options
+      attr_accessor :chef_config
+
+      def initialize(name, source_options, chef_config)
+        @name = name
+        @source_options = source_options
+        @chef_config = chef_config
+      end
+
+      def valid?
+        errors.empty?
+      end
+
+      def errors
+        error_messages = []
+
+        [:server, :policy_name].each do |key|
```

Comment:
We should let `policy_name:` default to the name, since it should almost always be the same thing.

Reply:
changed